### ISSUE TITLE
Error: Promise timed out

### ISSUE THREAD

**Issue by lucasoliveiraw00** [segmentio/analytics-react-native#982]
I'm facing a "Promise timed out" error in my application. Is anyone else having this issue?
  
![Captura de Tela 2024-08-01 às 21 22 52](https://github.com/user-attachments/assets/e0ac0e88-e4f8-45f0-8f13-9cc5acd2a32e)
![Captura de Tela 2024-08-01 às 21 29 34](https://github.com/user-attachments/assets/782b95f0-df1c-40de-b1b6-b0767128db82)

- `analytics-react-native` version: `2.19.2`
- Integrations versions (if used):
- React Native version: `0.73.8`
- iOS or Android or both? Android

**Comment by kip-west** [comment]
> I'm facing a "Promise timed out" error in my application. Is anyone else having this issue?
> 
> ![Captura de Tela 2024-08-01 às 21 22 52](https://private-user-images.githubusercontent.com/42048768/354420235-e0ac0e88-e4f8-45f0-8f13-9cc5acd2a32e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NzY5OTIsIm5iZiI6MTcyMjk3NjY5MiwicGF0aCI6Ii80MjA0ODc2OC8zNTQ0MjAyMzUtZTBhYzBlODgtZTRmOC00NWYwLThmMTMtOWNjNWFjZDJhMzJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDIwMzgxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlNTQ5MGZlNTAyODUyNDQwMmI1YjIwNmU5MmI2NDEwZTYwZjU1Y2Q1OGRmNjBkODQwYWMzNTBhMmZjZjg0MTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hRicdN2GRpIL1HIOJQAZndgz22aVuZZ_iG_xUJD8Thk) ![Captura de Tela 2024-08-01 às 21 29 34](https://private-user-images.githubusercontent.com/42048768/354421041-782b95f0-df1c-40de-b1b6-b0767128db82.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NzY5OTIsIm5iZiI6MTcyMjk3NjY5MiwicGF0aCI6Ii80MjA0ODc2OC8zNTQ0MjEwNDEtNzgyYjk1ZjAtZGYxYy00MGRlLWIxYjYtYjA3NjcxMjhkYjgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDIwMzgxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNlY2FiOTE5NTExZWNjNDVlYTQxNjg3ODNhYTYwYmFmNzAyMDljNDA1YmVjMjM5ODc4YmNiZGExNzRhYTdjYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gD7fDWb3u_dObnHebIFauWBhc3LCTIVq-9SHxtujxWU)
> 
> * `analytics-react-native` version: `2.19.2`
> * Integrations versions (if used):
> * React Native version: `0.73.8`
> * iOS or Android or both? Android

I am also having this issue!

`analytics-react-native version`: `2.19.2`
Integrations versions (if used):
React Native version: `0.73.6`
iOS or Android or both? both

**Comment by bit-scrambler** [comment]
Same for me.

**Comment by efstathiosntonas** [comment]
Duplicate of: https://github.com/segmentio/analytics-react-native/issues/954 which shouldn't be closed. 

This issue is around for 2 months and no one from segment team cares to fix.

**Comment by amkiran** [comment]
I am having the issue too. Any updates?

**Comment by smfunder** [comment]
@alanjcharles I've updated Segment to all the latest as possible an unfortunately I'm still receiving the promise timeout issue. Looks like to be on the utils file and I found it is only called on the QueueFlushingPlugin.
To me it is happening all the time.

I have:
    "@segment/analytics-react-native": "^2.19.5",
    "@segment/sovran-react-native": "^1.1.2",

React Native: 0.73.1

And this is my Sentry log. I'm now muting this error from Sentry because it is too noisy.

![Captura de pantalla 2024-10-07 a la(s) 2 42 58 p  m](https://github.com/user-attachments/assets/45e8f03d-e94a-46bb-9df6-c5e58a6f6a75)
